google-cloud-storage/src/main/java/com/google/cloud/storage/StorageImpl.java

```diff
@@ -132,24 +133,29 @@ public Blob create(BlobInfo blobInfo, byte[] content, BlobTargetOption... option
         .setCrc32c(BaseEncoding.base64().encode(
             Ints.toByteArray(Hashing.crc32c().hashBytes(content).asInt())))
         .build();
-    return create(updatedInfo, new ByteArrayInputStream(content), options);
+    return internalCreate(updatedInfo, content, options);
   }
 
   @Override
+  @Deprecated
   public Blob create(BlobInfo blobInfo, InputStream content, BlobWriteOption... options) {
     Tuple<BlobInfo, BlobTargetOption[]> targetOptions = BlobTargetOption.convert(blobInfo, options);
-    return create(targetOptions.x(), content, targetOptions.y());
+    StorageObject blobPb = targetOptions.x().toPb();
+    Map<StorageRpc.Option, ?> optionsMap = optionMap(targetOptions.x(), targetOptions.y());
+    InputStream inputStreamParam = firstNonNull(content, new ByteArrayInputStream(EMPTY_BYTE_ARRAY));
+    // retries are not safe when the input is an InputStream, so we can't retry.
+    return Blob.fromPb(this, storageRpc.create(blobPb, inputStreamParam, optionsMap));
```

Comment:
Another possibility would be to use InputStream.mark()/InputStream.reset() methods combination. Though even with those methods it is hard to provide robust implementation. At the same time with those methods we can reuse inputstream version of create() method for the byte[] version (ByteArrayInputStream must obviously support mark()/reset() in the most efficient way possible).